src/Elements/Input.react.js

```diff
@@ -176,12 +176,14 @@ class Input extends Component {
 
     _onChange(event) {
         const { value } = this.state;
-        const { max, min, onChange } = this.props;
+        const { max, min, onChange, allowEmpty } = this.props;
         const type = this._getType();
         let newValue = event.target.value;
 
         if (type === 'number' && _.isNumber(max) || _.isNumber(min)) {
```

Comment:
These two nested if statements are pretty confusing.  Can we combine into one and simplify?

Reply:
Fixed